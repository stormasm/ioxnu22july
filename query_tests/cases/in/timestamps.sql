-- Timestamp printing / output testss
-- IOX_SETUP: OneMeasurementRealisticTimes

-- Expect the timestamp output to be formatted correctly (with `Z`)
SELECT * from cpu;
-- explicit offset format
SELECT * FROM cpu WHERE time  > to_timestamp('2021-07-20 19:28:50+00:00');
-- Use RCF3339 format
SELECT * FROM cpu WHERE time  > to_timestamp('2021-07-20T19:28:50Z');
--use cast workaround
SELECT * FROM cpu WHERE
  CAST(time AS BIGINT) > CAST(to_timestamp('2021-07-20T19:28:50Z') AS BIGINT);
