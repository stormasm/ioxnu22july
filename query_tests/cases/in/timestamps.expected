-- Test Setup: OneMeasurementRealisticTimes
-- SQL: SELECT * from cpu;
+--------+----------------------+------+
| region | time                 | user |
+--------+----------------------+------+
| west   | 2021-07-20T19:28:50Z | 23.2 |
| west   | 2021-07-20T19:30:30Z | 21   |
+--------+----------------------+------+
-- SQL: SELECT * FROM cpu WHERE time  > to_timestamp('2021-07-20 19:28:50+00:00');
+--------+----------------------+------+
| region | time                 | user |
+--------+----------------------+------+
| west   | 2021-07-20T19:30:30Z | 21   |
+--------+----------------------+------+
-- SQL: SELECT * FROM cpu WHERE time  > to_timestamp('2021-07-20T19:28:50Z');
+--------+----------------------+------+
| region | time                 | user |
+--------+----------------------+------+
| west   | 2021-07-20T19:30:30Z | 21   |
+--------+----------------------+------+
-- SQL: SELECT * FROM cpu WHERE CAST(time AS BIGINT) > CAST(to_timestamp('2021-07-20T19:28:50Z') AS BIGINT);
+--------+----------------------+------+
| region | time                 | user |
+--------+----------------------+------+
| west   | 2021-07-20T19:30:30Z | 21   |
+--------+----------------------+------+
